src/solver.jl

```diff
@@ -88,14 +95,30 @@ function solve(
 
         while kkt_error > ϵ && inner_iters < max_inner_iters
             total_iters += 1
-            # Compute the Newton step.
-            # TODO: Can add some adaptive regularization.
+
+            # Compute the (regularized) Newton step.
             # TODO: use a linear operator with a lazy gradient computation here.
-            mcp.F!(F, x, y, s; θ, ϵ)
-            mcp.∇F_z!(∇F, x, y, s; θ, ϵ)
-            linsolve.A = ∇F + tol * I
+            if regularize_linear_solve === :internal
+                mcp.F!(F, x, y, s; θ, ϵ, η = 0.0)
+                mcp.∇F_z!(∇F, x, y, s; θ, ϵ, η)
+            else
+                mcp.F!(F, x, y, s; θ, ϵ)
+                mcp.∇F_z!(∇F, x, y, s; θ, ϵ)
+            end
+
+            if regularize_linear_solve === :identity
```

Comment:
This could be folded into a `elseif` of above

Reply:
I considered that, but thought the logic was simpler this way. Basically, for `:internal` mode you need to regularize `∇F` directly when computed in place, and for both `:identity` and `:none` you compute it without the extra parameter.  Then, for `:identity` only do you modify it when passing to the linear solver by adding an explicit scaled identity (if `∇F` is square).